### PR TITLE
Additions to TopTabView for customization options

### PR DESCRIPTION
**Pull request by bozidarsevo**
## Description

As discussed in https://github.com/woocommerce/woocommerce-ios/pull/14182#issuecomment-2429101146 we agreed that it would be better to add additions to `TopTabView` instead of having a new UI component, so that we can reuse that one in Carrier packages list (with some customizations).

This PR adds those options/additions but keeps the setup for code that currently uses `TopTabView`.

Small disclaimer: branch name has some parts from another nonrelated branch (copy-paste issue :) )

## Testing information

Currently only `WooPaymentsDepositsOverviewView` uses `TopTabView` so only manual testing that should be done is to check if that component still works ok and looks the same.

## Screenshots

`TopTabView` previews:

| Default Light    | Default Dark |
| -------- | ------- |
| Screenshot 2024-10-22 at 15 22 18 | Screenshot 2024-10-22 at 15 22 42 |

| Customized Light    | Customized Dark |
| -------- | ------- |
| Screenshot 2024-10-22 at 15 23 03 | Screenshot 2024-10-22 at 15 23 22 |

`WooPaymentsDepositsOverviewView` expanded preview (no change):

| Trunk    | This PR |
| -------- | ------- |
| Screenshot 2024-10-22 at 15 29 08 | Screenshot 2024-10-22 at 15 27 19 |

---
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/229">20.9</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/ViewRelated/ReusableViews/SwiftUI Components/TopTabView.swift [hunk 1]
```diff
@@ -244,6 +303,62 @@ struct ContentView_Previews: PreviewProvider {
...
+                   tabsContainerHorizontalPadding: nil,
+                   unselectedStateColor: .secondary,
+                   selectedTabIndicatorHeight: 3.0,
+                   tabPadding: 0,
+                   tabsNameFont: Font.subheadline.bold(),
+                   tabItemContentHorizontalPadding: 16.0,
+                   tabItemContentVerticalPadding: 9.0)
+            .previewLayout(.sizeThatFits)
+            .preferredColorScheme(.dark)
+            .previewDisplayName("Carrier Packages Dark Style")
```
Comment (bozidarsevo) [inline WooCommerce/Classes/ViewRelated/ReusableViews/SwiftUI Components/TopTabView.swift:361]: Added for easier preview of different styles and customizations

**Review by rachelmcr — APPROVED** [review]
Nice! These changes look great, and the previews are really useful for seeing how the customization options work 👍

**Pull request merged**